hbase-common/src/main/java/org/apache/hadoop/hbase/nio/MultiByteBuff.java

```diff
@@ -53,6 +55,19 @@
   private int markedItemIndex = -1;
   private final int[] itemBeginPos;
 
+  private Iterator<ByteBuffer> buffsIterator = new Iterator<ByteBuffer>() {
+    @Override
+    public boolean hasNext() {
+      return curItemIndex <= limitedItemIndex && items[curItemIndex].hasRemaining();
```

Comment:
One concern here, assume the following steps:
1. currentByteBuffer = it#next();
2. it#hasNext() must return true;
3. we write the current ByteBuffer &  fullfill it, then its hasRemaining will be false ....
4. it#hasNext() will return false because of the currBuff.hasRemaining is false.

Say for the same ByteBuffer, the it#hasNext() will return different value. In a general sense, if we don't have any Next(), then the it#hasNext() should always have the same return value, I mean ? 

Maybe we can remove the `&& items[curItemIndex].hasRemaining()` here, can let the outside logic hande it ?

Reply:
how about change it like this
`curItemIndex < limitedItemIndex || (curItemIndex == limitedItemIndex && items[curItemIndex].hasRemaining())`